cpp/src/link_analysis/pagerank.cu

```diff
@@ -183,44 +183,30 @@ template int pagerankSolver<int, double> (  int n, int e, int *cscPtr, int *cscI
         int *prsVtx,  double *prsVal, int prsLen, bool has_personalization,
         double alpha, double *a, bool has_guess, float tolerance, int max_iter, double * &pagerank_vector, double * &residual);
 
-template <typename WT>
+template <typename VT,typename WT>
 void pagerank_impl (Graph *graph,
-                      gdf_column *pagerank,
-                      gdf_column *personalization_subset, gdf_column *personalization_values,
-                      float alpha = 0.85,
-                      float tolerance = 1e-4, int max_iter = 200,
-                      bool has_guess = false) {
+                    WT* pagerank,
+                    size_t personalization_subset_size=0, VT* personalization_subset=nullptr,WT* personalization_values=nullptr,
+                    float alpha = 0.85,
+                    float tolerance = 1e-4, int max_iter = 200,
+                    bool has_guess = false) {
+
   bool has_personalization = false;
-  int *prsVtx = nullptr;
-  WT  *prsVal = nullptr;
   int prsLen = 0;
-  CUGRAPH_EXPECTS((personalization_subset == nullptr) == (personalization_values == nullptr), "Invalid API parameter");
-  if (personalization_subset != nullptr) {
-    has_personalization = true;
-    prsVtx = reinterpret_cast<int*>(personalization_subset->data);
-    prsVal = reinterpret_cast<WT* >(personalization_values->data);
-    prsLen = reinterpret_cast<int >(personalization_subset->size);
-    CUGRAPH_EXPECTS(pagerank->dtype == personalization_values->dtype, "Invalid API parameter");
-    CUGRAPH_EXPECTS(personalization_subset->dtype == GDF_INT32, "Unsupported data type");

```

Comment:
Do we need to forgo these checks? Or should they still be kept as sanity checks verifying dtypes, in case someone uses the CPP api

Reply:
Yes. I had mostly C++ users in mind for those. It is also helpful for cugrah's python developer and debugging as it will tell directly if they passed the wrong data type.